src/view.rs

```diff
@@ -545,6 +547,33 @@ pub trait LinkMut: LinkView + PortMut {
     /// Unlinks all connections to the `port`. If the port was connected,
     /// returns one of the ports it was connected to.
     fn unlink_port(&mut self, port: PortIndex) -> Option<Self::LinkEndpoint>;
+
+    /// Inserts another graph into this graph.
+    ///
+    /// Returns a map from the old node indices to the new node indices.
+    fn insert_graph(
+        &mut self,
+        other: &impl LinkView,
+    ) -> Result<HashMap<NodeIndex, NodeIndex>, LinkError> {
+        self.reserve(other.node_count(), other.port_count());
+        let mut rekeys = HashMap::with_capacity(other.node_count());
+        for old in other.nodes_iter() {
+            let new = self.add_node(other.num_inputs(old), other.num_outputs(old));
+            rekeys.insert(old, new);
+            for (from, to) in other.all_links(old) {
```

Comment:
I think that `all_links` returns all links (incoming and outgoing) from `old`, but on the line below it looks like you're assuming they're all outgoing links from old. And surely the `continue` trick only works if you're considering incoming *and* outgoing links?

It seems like there's at least some assumption that nodes will be iterated in reverse chronological order which isn't documented